File: frontend/src/model/Pet.ts
import {Shelter} from "./Shelter.ts";

export type Pet = {
    id: string,
    name: string,
    type: string,
    species: string,
    gender: string,
    castrated: string,
    age: string,
    description: string,
    images: string[],
    shelter: Shelter
}

File: backend/src/main/java/com/github/pschyl/backend/model/Pet.java
package com.github.pschyl.backend.model;

import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.util.List;

@Data
@RequiredArgsConstructor
public class Pet {
    private final String id;
    private final String name;
    private final String type;
    private final String species;
    private final String gender;
    private final String castrated;
    private final String age;
    private final String description;
    private final List<String> images;
    private final Shelter shelter;
}
